src/mesh/coordinates.cxx

```diff
@@ -900,32 +900,27 @@ int Coordinates::calcCovariant() {
 
   auto a = Matrix<BoutReal>(3, 3);
 
-  for (int jx = 0; jx < localmesh->LocalNx; jx++) {
-    for (int jy = 0; jy < localmesh->LocalNy; jy++) {
-      // set elements of g
-      a(0, 0) = g11(jx, jy);
-      a(1, 1) = g22(jx, jy);
-      a(2, 2) = g33(jx, jy);
-
-      a(0, 1) = a(1, 0) = g12(jx, jy);
-      a(1, 2) = a(2, 1) = g23(jx, jy);
-      a(0, 2) = a(2, 0) = g13(jx, jy);
-
-      // invert
-      if (invert3x3(a)) {
-        output_error.write("\tERROR: metric tensor is singular at (%d, %d)\n", jx, jy);
-        return 1;
-      }
+  BOUT_FOR_SERIAL(i, g11.getRegion(region)) {
```

Comment:
Is there a reason for this to be `BOUT_FOR_SERIAL` rather than `BOUT_FOR`? Not performance critical, since this loop is only needed for initialization - i'm just curious.

Reply:
Yes, there's a `return` (if `invert3x3` fails) which is not allowed inside an OpenMP region